src/main/java/com/monitorjbl/plugins/UserUtils.java

```diff
@@ -19,44 +23,44 @@ public UserUtils(UserService userService) {
     this.userService = userService;
   }
 
-  public List<User> dereferenceUsers(Iterable<String> users) {
-    List<User> list = newArrayList();
-    for(String u : users) {
-      Optional<User> user = getUserByName(u);
-      if(user.isPresent()) {
-        list.add(user.get());
-      }
+  public List<User> dereferenceUsers(Collection<String> usernames) {
+    Set<String> distinctNames = new HashSet<>(usernames);
+    if (distinctNames.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    Set<ApplicationUser> usersByName = userService.getUsersByName(distinctNames);
+    if (usersByName.isEmpty()) {
+      return Collections.emptyList();
     }
-    return list;
-  }
 
-  public Optional<User> getUserByName(String username) {
-    ApplicationUser user = userService.getUserByName(username);
-    return user == null ? Optional.empty() : Optional.of(new User(user.getName(), user.getDisplayName()));
+    return usersByName.stream()
+        .map(user -> new User(user.getName(), user.getDisplayName()))
+        .collect(toList());
   }
 
   public String getUserDisplayNameByName(String username) {
-	    ApplicationUser user = userService.getUserByName(username);  
-	    return user.getDisplayName() == null ? username : user.getDisplayName();
+    return ofNullable(userService.getUserByName(username))
```

Comment:
This change guards against `getUserByName` returning `null` (It's marked `@Nullable` in our documentation)

Reply:
Actually, from looking at the open issues for the plugin, this change will fix #69 (Or, at least, will prevent the `NullPointerException` from that issue from occurring)